Assets/Scripts/UI/Canvas/ColorPicker/ColorPicker.cs

```diff
@@ -0,0 +1,70 @@
+using UnityEngine;
+
+using Framework.Enums;
+
+namespace UI.Canvas.ColorPicker
+{
+    public class ColorPicker : MonoBehaviour
+    {
+        [SerializeField] private LineRenderer routeLineRenderer;
+    
+        private void Start()
+        {
+            gameObject.SetActive(false);
+        }
+
+        /// <summary>
+        /// Checks which RouteColors enum goes with which color
+        /// </summary>
+        /// <param name="lineColor"> The RouteColors enum that gets grabbed from the button that's clicked </param>
+        /// <returns> Returns the color that is used for the ChangeColor function </returns>
+        public Color SetColor(RouteColors lineColor)
+        {
+            Color newColor = new Color();
+            
+            switch (lineColor)
+            {
+                case RouteColors.RED:
+                    newColor = new Color(1f,0f,0f);
+                    break;
+                case RouteColors.GREEN:
+                    newColor = new Color(0f, 1f, 0f);
+                    break;
+                case RouteColors.BLUE:
+                    newColor = new Color(0f, 0f, 1f);
+                    break;
+                case RouteColors.YELLOW:
+                    newColor = new Color(1f, 1f, 0f);
+                    break;
+                case RouteColors.MAGENTA:
+                    newColor = new Color(1f, 0f, 1f);
+                    break;
+                case RouteColors.ORANGE:
+                    newColor = new Color(1f, 0.5f, 0f);
+                    break;
+            }
+            
+            return newColor;
+        }
+        
+        /// <summary>
+        /// Changes the line renderer's line color to a new color
+        /// </summary>
+        /// <param name="newColor"> The color to set the line to </param>
+        public void ChangeColor(ColorChange colorChange)
+        {
+            routeLineRenderer.startColor = SetColor(colorChange.ColorToChange);
+            routeLineRenderer.endColor = SetColor(colorChange.ColorToChange);
+        }
```

Comment:
```suggestion
        public void ChangeColor(ColorChange colorChange)
        {
            Color targetColor = SetColor(colorChange.ColorToChange);
            routeLineRenderer.startColor = targetColor;
            routeLineRenderer.endColor = targetColor;
        }
```
Nu roep je de `SetColor` maar 1 keer aan ipv 2 keer. Nu checkt het ook maar 1 keer die switch case en (zonder vorie feedback applied) maak je 4 keer een color varible aan.